src/Compute/Compute/VirtualMachine/Config/SetAzureVMOperatingSystemCommand.cs

```diff
@@ -306,11 +306,20 @@ public override void ExecuteCmdlet()
         {
             if (this.VM.OSProfile == null)
             {
+                if (this.Credential == null)
```

Comment:
Is Credential always needed? Or only when OSProfile is not provided? 

Is it because if OSProfile is NOT null, credentials could be in there already?